src/features/recs-pathways/widget-parameters.type.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * A comma-separated list of fields to be sent in the request.
+ *
+ * Alternatively, you may configure the fields in the Widget Configurator in the Dashboard instead. This parameter is required if not sent via the Dashboard.
+ */
+export type fields = string;
+
+/**
+ * The filter parameter applies a faceted filter to the returned products, searching for products that fit your parameter values. Any facet that you want to filter must be in your feed.
+ *
+ * Attributes must be enabled and mapped by Bloomreach. Let your Bloomreach representative know which attributes in your content feed you want to apply as filters to search results.
+ *
+ * You can filter results based on numeric ranges. For example, &filter=(price:["100" TO "*"]). Please note that a negative filter on ranges is not supported.
+ *
+ * To provide multiple filters, send multiple filter parameters. For example, &filter=(price:["*" TO "100"])&filter=(color_groups: ("blue"))
+ */
+export type filter = string;
+
+/**
+ * Your site's category ID.
+ *
+ * The example value shown here, pnb160000000000, is included for your convenience to send a request with Try It.
+ */
+export type cat_id = string;
```

Comment:
Just curious, what's the purpose of using this kind of "type alias" instead of making cat_id just of type string?  Is it for anticipating potential future type changes on cat_id, such as we may include number or null?

Reply:
The reason is to keep it similar to the other properties defined in the other type files. And the reason I do that is because it turned out a these properties are the same in different APIs, yet in some places they might be required where in others they are not, and I didn't want to duplicate the explanation comment above (thee for IDE completion purposes). I tried grouping the parameters in groups like 'AccountParams' or 'TrackingParams' but then ran into cases where for example the search APIs use account_id and domain_key, but recommendations only use account_id etc.

So I figured lets break it down to atomic level, similar to the ['component' definitions](https://code.bloomreach.com/engineering/discovery/recommendations/frodo/-/blob/master/app/api_specification/spec.yaml?ref_type=heads#L21) in the OpenAPI spec.

Reply:
Thanks for the explanation.  Makes sense!